SimulationAgent.Test/SimulationRunnerTest.cs

```diff
@@ -0,0 +1,155 @@
+        private readonly Mock<IDeviceTelemetryActor> deviceTelemetryActor;
+        private readonly Mock<UpdateDeviceState> updateDeviceStateLogic;
+        private readonly SimulationRunner target;
+
+        public SimulationRunnerTest(ITestOutputHelper log)
+        {
+            this.ratingConfig = new Mock<IRateLimitingConfig>();
+            this.logger = new Mock<ILogger>();
+            this.deviceModels = new Mock<IDeviceModels>();
+            this.deviceModelsOverriding = new Mock<IDeviceModelsGeneration>();
+            this.devices = new Mock<IDevices>();
+            this.simulations = new Mock<ISimulations>();
+            this.factory = new Mock<IFactory>();
+            this.deviceStateActors = new Mock<IDictionary<string, IDeviceStateActor>>();
+            this.deviceStateActor = new Mock<IDeviceStateActor>();
+            this.deviceConnectionActor = new Mock<IDeviceConnectionActor>();
+            this.deviceTelemetryActor = new Mock<IDeviceTelemetryActor>();
+            this.updateDeviceStateLogic = new Mock<UpdateDeviceState>();
+
+            this.target = new SimulationRunner(
+                this.ratingConfig.Object,
+                this.logger.Object,
+                this.deviceModels.Object,
+                this.deviceModelsOverriding.Object,
+                this.devices.Object,
+                this.simulations.Object,
+                this.factory.Object);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ActiveCountInitToZero()
+        {
+            // Act
+            var result = this.target.GetActiveDevicesCount();
+
+            // Assert
+            Assert.Equal(0, result);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void AbleToGetActiveDeviceCount()
+        {
+            // Arrange
+            const int DEVICE_COUNT = 7;
+            var models = new List<DeviceModelRef>
+            {
+                new DeviceModelRef { Id = "01", Count = DEVICE_COUNT }
+            };
+
+            var simulation = new SimulationModel
+            {
+                Id = "1",
+                Created = DateTimeOffset.UtcNow.Subtract(TimeSpan.FromDays(10)),
+                Modified = DateTimeOffset.UtcNow.Subtract(TimeSpan.FromDays(10)),
+                ETag = "ETag0",
+                Enabled = false,
+                Version = 1,
+                DeviceModels = models
+            };
+
+            SetupSimulationReadyToStart();
+
+            // Act
+            this.target.Start(simulation);
+            var result = this.target.GetActiveDevicesCount();
+
+            // Assert
+            Assert.Equal(DEVICE_COUNT, result);
+        }
+
+        /*
+         * Setup deviceModel, devicesIds, simulations, devices, deviceStateActor,
+         * deviceConnectionActor, deviceTelemetryActor for simulationRunner
+         */
+        private void SetupSimulationReadyToStart()
```

Comment:
split in multiple private methods, giving them a readable name, so you can drop the comments